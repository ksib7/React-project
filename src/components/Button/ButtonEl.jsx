import React from "react";
import Box from "@mui/material/Box";
import Button from "@mui/material/Button";
import PropTypes from "prop-types";
import "./Button.scss";

export const ButtonEl = ({ children, onButtonClick }) => {
  return (
    <Box className="flex" mt={2}>
      <Button variant="contained" type="submit" onClick={onButtonClick}>
        {children}
      </Button>
    </Box>
  );
};

ButtonEl.propTypes = {
  onButtonClick: PropTypes.func,
};
