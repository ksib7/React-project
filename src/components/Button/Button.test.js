import React from "react";
import { render, screen } from "@testing-library/react";
import { ButtonEl as Button } from "./ButtonEl";
import "@testing-library/jest-dom";

describe("Button", () => {
  it("Component is rendered", () => {
    render(<Button />);
  });

  it("Button contains children prop", () => {
    render(<Button />);
    expect(screen.getByTestId("button")).toHaveProperty("children");
  });

  it("Button is disabled", () => {
    render(<Button disabled />);
    const button = screen.getByRole("button");
    expect(button).toBeDisabled();
  });

  it("render button component with snapshot (remember the mark up)", () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });
});
