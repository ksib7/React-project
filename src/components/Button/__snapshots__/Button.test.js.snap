// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render button component with snapshot (remember the mark up) 1`] = `
<DocumentFragment>
  <div
    class="flex MuiBox-root css-1yuhvjn"
  >
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      data-testid="button"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`Button render button component with snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex MuiBox-root css-1yuhvjn"
  >
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      data-testid="button"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
